onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -16,47 +18,50 @@
 #include "core/framework/mlvalue_name_idx_map.h"
 #include "core/framework/sequential_execution_plan.h"
 #include "core/framework/session_state.h"
-#include "core/framework/tensorutils.h"
 #include "core/framework/tensorprotoutils.h"
 #include "core/framework/utils.h"
+#include "core/framework/mem_buffer.h"
 
 namespace onnxruntime {
 
 static common::Status SaveMLValueNameIndexMapping(const GraphViewer& graph_viewer,
                                                   MLValueNameIdxMap& mlvalue_name_idx_map,
                                                   const logging::Logger& logger);
 
-using SaveTensorFunc = std::function<void(int idx, const onnxruntime::MLValue&)>;
-
-static common::Status SaveInitializedTensors(const onnxruntime::Graph& graph,
-                                             bool enable_memory_pattern,
+// T should have signature of '(int idx, const onnxruntime::MLValue& value, const OrtDeleter& d) -> Status'
+template <typename T>
```

Comment:
If there's a fixed signature for T, can you instead use std::function<Status(int, const MLValue&, const OrtDeleter&)> for the param?

Reply:
Template has better perf. It can avoid the conversion from a lambda(or function pointer) into a std::function.  And, when it is running, if you take a look at the call stack,  the call stack is much simpler.  I think std::function is mainly for async uses, that you want to store this object at somewhere and invoke it at a later time.

Reply:
Ahh. Good to know. Thanks.